.github/workflows/miden.yml

```diff
@@ -0,0 +1,341 @@
+#             sudo systemctl start miden-node
+#             sudo systemctl enable miden-faucet
+#             sudo systemctl start miden-faucet
+# EOF
+
+#       - name: SSH into EC2 devnet and upgrade miden-node
+#         if: needs.check-crates.outputs.update_needed_2 == 'true'
+#         run: |
+#           ssh -i polygon-miden-ec2.pem -o "StrictHostKeyChecking no" ec2-user@3.255.177.27 << 'EOF'
+#             # Stop the miden-node service
+#             sudo systemctl stop miden-node
+#             sudo systemctl stop miden-faucet
+
+#             # Remove previous data
+#             rm -rf ~/miden-node; mkdir ~/miden-node
+#             rm -rf ~/miden-faucet; mkdir ~/miden-faucet
+
+#             # Install latest versions of miden-node and miden-faucet
+#             cargo install miden-node miden-faucet --locked --features testing
+
+#             # Initialize the services again
+#             miden-node init -c ~/miden-node/miden-node.toml -g ~/miden-node/genesis.toml
+#             miden-node make-genesis -i ~/miden-node/genesis.toml -o ~/miden-node/genesis.dat
+#             miden-faucet init -c ~/miden-faucet/miden-faucet.toml
+
+#             # Configure systemd service
+#             sudo tee /etc/systemd/system/miden-node.service > /dev/null <<EOT
+#             sudo tee /etc/systemd/system/miden-faucet.service > /dev/null <<EOT
+
+#             [Unit]
+#             Description=Miden Node
+#             After=network.target
+
+#             [Service]
+#             ExecStart=/home/ec2-user/.cargo/bin/miden-node start -c miden-node.toml node
+#             ExecStart=/home/ec2-user/.cargo/bin/miden-faucet start -c miden-faucet.toml
+#             Restart=on-failure
+#             User=ec2-user
+#             WorkingDirectory=/home/ec2-user/miden-node
+#             WorkingDirectory=/home/ec2-user/miden-faucet
+
+#             Environment="RUST_LOG=info"
+#             LimitNOFILE=10000
+
+#             [Install]
+#             WantedBy=multi-user.target
+#             EOT
+
+#             sudo systemctl daemon-reload
+#             sudo systemctl enable miden-node
+#             sudo systemctl start miden-node
+#             sudo systemctl enable miden-faucet
+#             sudo systemctl start miden-faucet
+# EOF
+
+      - name: SSH into EC2 and upgrade miden-node
+        run: |
+          ssh -i miden-test.pem -o "StrictHostKeyChecking no" ec2-user@{{ env.AWS_IP }} << 'EOF'
+            # Stop the miden-node service
+            sudo systemctl stop miden-node
+            sudo systemctl stop miden-faucet
+
+            # Remove previous data
+            # rm -rf /home/ec2-user/miden-node
+            # rm -rf /home/ec2-user/miden-faucet
+
+            rm -rf ~/miden-node; mkdir ~/miden-node
+            rm -rf ~/miden-faucet; mkdir ~/miden-faucet
+
+            # Install latest versions of miden-node and miden-faucet
+            cargo install miden-node miden-faucet --locked --features testing
```

Comment:
This needs to be done as separate steps because it depends on the network type.

```
- name: Install testnet node
  if: env.network == 'testnet'
  run: |
    ssh -i miden-test.pem -o "StrictHostKeyChecking no" ec2-user@{{ env.AWS_IP }} << 'EOF'
    cargo install miden-node miden-faucet --locked --features testing
    EOF

- name: Install devnet node
  if: env.network == 'devnet'
  run: |
    ssh -i miden-test.pem -o "StrictHostKeyChecking no" ec2-user@{{ env.AWS_IP }} << 'EOF'
    cargo install --git https://github.com/0xPolygonMiden/miden-node miden-node --branch next --bin miden-node --locked --features testing
    cargo install --git https://github.com/0xPolygonMiden/miden-node miden-faucet --branch next --bin miden-faucet --locked --features testing
    EOF
```